### DISCUSSION TITLE
Script main.py (or boot.py) doesn't properly start when module reset

### DISCUSSION THREAD

**Discussion by eljots** [micropython/micropython#11411] [Using MicroPython]
Hi, as this is my first post here I'd like to welcome and wish all you the best.
Again, as I'm just starting with ESP and uPython my problem probably is obvious but so far have not found solution :-(
I try to start my python script at boot (or reset) using boot.py or main.py but receive such message when load manually eg. main.py:

download ok
exec(open('main.py').read(),globals())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<string>", line 11, in <module>
NameError: name 'I2C' isn't defined

My boot.py (or main.py - I've tried both) is as follow:

import machine
import sys
import ds1302
from machine import Pin, SoftI2C
from lcd_api import LcdApi
from i2c_lcd import I2cLcd
from time import sleep
I2C_ADDR = 0x27
totalRows = 2
totalColumns = 16
_**i2c = I2C(scl=Pin(5), sda=Pin(4), freq=500000)       #initializing the I2C method for ESP8266**_
lcd = I2cLcd(i2c, I2C_ADDR, totalRows, totalColumns)
............
............

Files lcd_api.py and i2c_lcd.py which mange I2C LCD module are present on ESP8266. 
Using uPyCraft v.1 to load (and run) scripts on ESP8266.
Interesting, when manually, prior to the main.py script with mentioned code, first load lcd_api.py and i2c_lcd.py files, then everything working fine. When module is restarted with reset switch then nothing on the screen.
Any ideas?

eljots

**Comment by robert-hh** [comment]
In line 4 you imported SoftI2C instead of I2C. Therefor I2C is not found.

**Comment by eljots** [comment]
Hi robert-hh!

Great advice, you are right, when corrected as suggested working fine.
I've found code example in net as:

.................
from machine import Pin, SoftI2C
................
................
#i2c = SoftI2C(scl=Pin(22), sda=Pin(21), freq=10000)     #initializing the I2C method for ESP32
i2c = I2C(scl=Pin(5), sda=Pin(4), freq=500000)       #initializing the I2C method for ESP8266
...............

Probably, author tried code for ESP32 only and forgot about importing I2C when ESP8266 run. I was misled when loading i2c_lcd.py first before main.py caused proper function of code.
It's  learning for me to be more critical about code from net.
Apparently, rebooting starts script main.py without errors and proper code is as follow:
...............
from machine import Pin, I2C
................
i2c = I2C(scl=Pin(5), sda=Pin(4), freq=500000)       #initializing the I2C method for ESP8266
...............

Thanks again and have a good day!

eljots
